### PR TITLE
Update for 04-25 snapshot

### PR DESCRIPTION
**Pull request by MoridinBG**
Update for the latest Swift syntax in snapshot 04-25

### CODE REVIEW

File: Sources/Fluent/Model.swift [hunk 1]
```diff
@@ -19,15 +19,15 @@ extension Model {
...
+        try Query().delete(model: self)
     }
 
     public static func all() throws -> [Self] {
         return try Query().all()
     }
     
     public static func find(ids: Value...) throws -> [Self] {
-        let result: [Self] = try Query().filter("id", in: ids).all()
+		let result: [Self] = try Query().filter(field: "id", in: ids).all()
```
Comment (tannernelson) [inline Sources/Fluent/Model.swift]: `filter()` should be made to not require a label for `field`. 

`_ field`

File: Sources/Fluent/Model.swift [hunk 2]
```diff
@@ -9,7 +9,7 @@ public protocol Model {
 
 extension Model {    
     public func save() throws {
-        try Query().save(self)
+        try Query().save(model: self)
```
Comment (tannernelson) [inline Sources/Fluent/Model.swift]: Same as `filter`, `model` should not be a required label.

**Comment by tannernelson** [comment]
Let's try to avoid changing the public API if possible (ie, using `_` to make the labels not required).

Otherwise looks good, thanks!

**Comment by MoridinBG** [comment]
Makes sense. I had a feeling it was making some things too verbose.
Will change and resubmit.

**Comment by abiaad** [comment]
Would be awesome to accept this request 👍

**Comment by MoridinBG** [comment]
That last one was not supposed to go here, but I pushed to the wrong branch

File: Sources/Fluent/Query.swift [hunk 3]
```diff
@@ -149,7 +149,8 @@ public class Query<T: Model> {
         return self
     }
 
-    public func list(key: String) throws -> [Value] {
+    public func list(_ key: String) throws -> [Value] {
+		self.fields.append(key)
```
Comment (tannernelson) [inline Sources/Fluent/Query.swift]: The spacing doesn't match here.

**Comment by tannernelson** [comment]
Looks good. Just one small spacing fix.

**Comment by timominous** [comment]
@MoridinBG Sorry, I pushed the 05-03 update to this repo. Apparently, my origin was set to this repo instead of my own copy of the repo. The plucking logic seems to be valuable.

**Comment by MoridinBG** [comment]
It's all right. I haven't had much time to dedicate to Fluent recently. Maybe in a few days, as I have some ideas.
I will clone your fixes and resubmit just the plucking fix

**Comment by timominous** [comment]
@MoridinBG Awesome. I'll close this PR now.

**Pull request closed without merging**